model/variable.cc

```diff
@@ -415,8 +415,8 @@ bool VariableBase::visible() const
   auto g=group.lock();
   //toplevel i/o items always visible
   if ((!g || !g->group.lock()) && g==controller.lock()) return true;
-  // ensure pars and constants with invisible out wires are made invisible. for ticket 1275  
-  if (type()==constant || type()==parameter)
+  // ensure pars, constants and flows with invisible out wires are made invisible. for ticket 1275  
+  if ((type()==constant || type()==parameter) && !ports[0]->wires().empty())
```

Comment:
You need to work out why varTabDisplay is being set on parameters on their creation. I can't see the smoking gun by eye - you'll need to debug it.

Reply:
Let's discuss this some more. Could you please outline what you think the rules should be, and we walk through the consequences.

Reply:
Actually, on second thoughts, unwired parameters should return false on attachedToDefiningVar() being true, so I will merge this change. But that and the the next clause should apply for all items, not just constants and variables, so should be moved into Item::visible. Once done, line 425 is no longer relevant. Line 426 also needs to be moved into IntOp (avoiding the dynamic_cast).  My thought is that an IntOp->attachedToDefiningVar() should delegate to its intVar.

Reply:
Thanks for these tips :) I can get most of what you suggest to work, but not the IntOp. It defies my every attempt when I move the code to the IntOp class... I'll keep trying and  hopefully by tomorrow morning (my time) have something working properly...